cpp/src/arrow/CMakeLists.txt

```diff
@@ -249,97 +352,92 @@ set(ARROW_SRCS
     util/unreachable.cc
     util/uri.cc
     util/utf8.cc
-    util/value_parsing.cc
-    vendored/base64.cpp
-    vendored/datetime/tz.cpp
-    vendored/double-conversion/bignum.cc
-    vendored/double-conversion/bignum-dtoa.cc
-    vendored/double-conversion/cached-powers.cc
-    vendored/double-conversion/double-to-string.cc
-    vendored/double-conversion/fast-dtoa.cc
-    vendored/double-conversion/fixed-dtoa.cc
-    vendored/double-conversion/string-to-double.cc
-    vendored/double-conversion/strtod.cc)
-
-if(ARROW_JEMALLOC)
-  list(APPEND ARROW_SRCS memory_pool_jemalloc.cc)
-  set_source_files_properties(memory_pool_jemalloc.cc
-                              PROPERTIES SKIP_UNITY_BUILD_INCLUSION ON)
+    util/value_parsing.cc)
+if(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+  set_property(SOURCE util/io_util.cc
+               APPEND_STRING
+               PROPERTY COMPILE_FLAGS " -Wno-unused-macros ")
```

Comment:
Can we perhaps use pragmas inside `io_util.cc` to avoid this?

Reply:
Yes, we can do it.
I'll add a change for it.

Reply:
I removed this and no CI weren't failed. So it seems that we don't need this now.